crates/lang/ir/src/ir/attrs.rs

```diff
@@ -565,6 +566,13 @@ impl InkAttribute {
     }
 }
 
+fn non_hex_err<'a>(
+    meta: &'a syn::Meta,
+    pos: usize,
+) -> impl FnOnce(ParseIntError) -> syn::Error + 'a {
+    move |_| format_err_spanned!(meta, "encountered non-hex digit at position {}", pos)
```

Comment:
why `move |_|` ? this is counter intuitive imo. Please just return a `syn::Error` here as usual. the point of making it possible to call it without the `|x| foo` in the invocation is not worse than this.